src/coreComponents/mesh/generators/NewGhosting.cpp

```diff
@@ -0,0 +1,1173 @@
+  //
+  // In order to do that, we build the `missingIndicator` matrix, which is rectangular:
+  // - The number of columns is the number of graph nodes in the mesh graph.
+  // - The number of rows is the total number of graph nodes that are missing on ranks.
+  //   Note that the same quantity can be missing on multiple ranks, and that's handled.
+  // - The non-zero terms equal `1`, meaning that a given quantity (column) is missing on a given rank (row).
+  //
+  // Combining `missingIndicator` with the adjacency matrix which conveys a lot of connections information,
+  // we'll be able to create the final `missingIndices` matrix,
+  // with `range` and `domain` maps (MPI ranks ownerships and offsets) are appropriately defined
+  // such that the rows will be available to any ranks that need them (nothing more, nothing less).
+  //
+  // To get the `missingNodePos` matrix which will convey the ghosted nodes positions that are missing on the rank,
+  // we first create a specific `missingNodesIndicator` matrix, which is alike `missingIndicator` but only for the nodes.
+  // We could have used `missingIndicator` and ditched the superfluous information,
+  // but the node positions are reals, where the connections are integers.
+  // As long as we're using `Epetra`, where the type of matrix term is `double`, this makes no critical difference.
+  // But when we switch to `Tpetra`, where we can select the type of the matrix term (and therefore use integers where we need integers),
+  // this may have become an issue.
+  // So the work has been done to separate `missingIndicator` and `missingNodesIndicator`.
+  //
+  // `missingNodesIndicator` is a rectangular like `missingIndicator`:
+  // - The number of columns is the number of graph nodes in the mesh graph
+  //   that actually are physical nodes in the mesh.
+  // - The number of rows is the total number of graph nodes (that actually are physical nodes in the mesh) that are missing on ranks.
+  // - The non-zero terms equal `1`, meaning that a given quantity (column) is missing on a given rank (row).
+  std::size_t const numLocMissingCompound[2] = { std::size( notPresentIndices ), std::size( notPresentNodes ) };
+  std::size_t numGlbMissingCompound[2] = { 0, 0 };
+  MpiWrapper::allReduce( numLocMissingCompound, numGlbMissingCompound, 2, MPI_SUM );
+  std::size_t const & numLocMissingIndices = numLocMissingCompound[0];
+  std::size_t const & numLocMissingNodes = numLocMissingCompound[1];
+  Tpetra::global_size_t const numGlbMissingIndices = numGlbMissingCompound[0];
+  Tpetra::global_size_t const numGlbMissingNodes = numGlbMissingCompound[1];
+
+  std::size_t const numGlbNodes = gis.nodes.get() + 1;  // TODO Use strongly typed integers
+  std::size_t const numOwnedNodes = std::size( ownedNodesIdcs );  // TODO Use strongly typed integers
+
+  auto const missingIndicesMap = make_rcp< TriMap const >( numGlbMissingIndices, numGlbMissingIndices, TriGlbIdx{ 0 }, comm );
```

Comment:
Should the second argument here be numLocMissingIndices instead of the global

Reply:
Hmm, yes probably, I would have expected a failure then...
I guess my current test case is too simple.